code/modules/power/lighting.dm

```diff
@@ -967,14 +977,10 @@ ADMIN_INTERACT_PROCS(/obj/machinery/light, proc/broken, proc/admin_toggle, proc/
 				boutput(user, "That's not safe with the power on!")
 				return
 			if (candismantle)
-				boutput(user, "You begin to unscrew the fixture from the wall...")
+				boutput(user, "You begin to loosen the fixture's screws...")
 				playsound(src.loc, 'sound/items/Screwdriver.ogg', 50, 1)
-				if (!do_after(user, 2 SECONDS))
-					return
-				boutput(user, "You unscrew the fixture from the wall.")
-				var/obj/item/light_parts/parts = new /obj/item/light_parts(get_turf(src))
-				parts.copy_light(src)
-				qdel(src)
+				SETUP_GENERIC_PRIVATE_ACTIONBAR(user, src, 2 SECONDS, PROC_REF(uninstall_fixture),list(), src.icon, src.icon_state,\
```

Comment:
Why private?

Reply:
`do_after` doesn't show anything externally, so it felt like the "least changes" version of modifying the code. After reviewing the lighting code again, going to change this to match the installation which is not private.